Lib/_pyio.py

```diff
@@ -504,8 +504,11 @@ def nreadahead():
                 return 1
         if size is None:
             size = -1
-        elif not isinstance(size, int):
-            raise TypeError("size must be an integer")
+        else:
+            try:
+                size = size.__index__()
+            except AttributeError as err:
+                raise TypeError("an integer is required") from err
```

Comment:
Given how `size` gets used after this, there shouldn't be any need to reassign it by calling `__index__`. Similarly for the other changes.